src/ls.js

```diff
@@ -128,3 +149,16 @@ function _ls(options, paths) {
   return list;
 }
 module.exports = _ls;
+
+
+function ls_stat(path) {
+  var stats = fs.statSync(path);
+  // Note: this object will contain more information than .toString() returns
+  var long_stats = Object.create(stats);
+  long_stats.name = path;
```

Comment:
I see the benefit, but it could be slightly trickier that way. `ls -l` returns output with the relative path to the file, so that's what I'm opting for with this. If users need the specific name, they can use `path.basename()` to get it. If they need the full path, I think `path.resolve()` would do it.

I could use these functions to set the internal properties, but it's harder to use the full path and basename to get the relative path for the `toString()` method.